vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -348,7 +343,24 @@ void WinRTWebSocketResource::Connect(const Protocols &protocols, const Options &
   }
 
   m_connectRequested = true;
-  PerformConnect();
+
+  Uri uri{nullptr};
+  try {
+    uri = Uri{winrt::to_hstring(std::move(url))};
+  } catch (hresult_error const &e) {
+    if (m_errorHandler) {
+      m_errorHandler({HResultToString(e), ErrorType::Connection});
```

Comment:
>HResultToString [](http://example.com/codeflow?start=22&length=15)

(nit) although converting to string is nice here, it would still be good to propagate the actual HResult value into m_errorHandler as well.

Reply:
This was discussed before.

Keeping the HRESULT would be both overkill because only the error string needs to be propagated to the JS layer , and this would potentially break the abstraction (IWebSocketResource is CppWinRT and COM-agnostic).